src/reducers/projects.js

```diff
@@ -83,6 +83,21 @@ export function reduceRoot(stateIn, action) {
             projects.get(currentProjectKey),
           );
         }
+      case 'FOCUS_LINE':
+        {
+          const currentProjectKey =
+            stateIn.getIn(['currentProject', 'projectKey']);
+
+          if (isNil(currentProjectKey)) {
+            return stateIn;
+          }
+
+          return projects.updateIn(
+            [currentProjectKey, 'hiddenUIComponents'],
+            hiddenUIComponents =>
+              hiddenUIComponents.remove(`editor.${action.payload.language}`),
+          );
+        }
```

Comment:
Alternatively - is there a reason this has to be on its own and not be tacked on top of the existing `UNHIDE_COMPONENT` reducer like so?
```
    case 'FOCUS_LINE':
    case 'UNHIDE_COMPONENT':
      return state.updateIn(
        [action.payload.projectKey, 'hiddenUIComponents'],
        hiddenUIComponents =>
          hiddenUIComponents.delete(action.payload.componentName),
      ).setIn(
        [action.payload.projectKey, 'updatedAt'],
        action.meta.timestamp,
      );
```
You'd just have to change `action.payload.language` to `action.payload.componentName` (or vice versa) and consolidate how you're handling `projectKey`. You'd also get the benefit of the `updatedAt` getting properly updated (not sure if that was intentionally left out of the new proposed reducer case or not).